tools/loader/ModelCompiler.cpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Loader.h"
+
+#include "glow/Converter/TypeAToTypeBFunctionConverter.h"
+#include "glow/Graph/Nodes.h"
+#include "glow/Importer/Caffe2ModelLoader.h"
+#include "glow/Importer/ONNXModelLoader.h"
+
+#include "llvm/Support/CommandLine.h"
+
+using namespace glow;
+
+namespace {
+
+/// Model compiler options.
+llvm::cl::OptionCategory modelCompilerCat("Model Compiler Options");
+
+llvm::cl::list<std::string> modelInputs(
+    "model-input", llvm::cl::ZeroOrMore,
+    llvm::cl::desc(
+        "For ONNX models the inputs of the graph can be inferred   \n"
+        "automatically and hence this option is not mandatory.     \n"
+        "For Caffe2 models the graph definition does not contain   \n"
+        "the description of the inputs and hence must be provided  \n"
+        "explicitly using this option. One or more model inputs    \n"
+        "are provided using the following format:                  \n"
+        "   -model-input=<inputName1>,<inputType1>,<inputShape1>   \n"
+        "   -model-input=<inputName2>,<inputType2>,<inputShape2>   \n"
+        "   ....................................................   \n"
+        "For example we can can describe 2 inputs as:              \n"
+        "   -model-input=input_03_data,float,[1]                   \n"
+        "   -model-input=data_bias,int32,[1,32,32]                 \n"
+        "The supported types are: float, float16, int32, int64 and \n"
```

Comment:
Why only this subset of types is supported? May be we should allow for (almost) all types supported by Glow? E.g. quantized types could be interesting if ones uses pre-quantized models.

Reply:
I chose only these types because they are standard. Currently the models are quantized with Glow only internally (in the middle) while keeping float placeholders (inputs/outputs).
I can also add the quantized types, this would mean adding a format of the type:
-model-input=<name><type>,<shape>                                for non-quantized types
-model-input=<name>,<type>,<scale>,<offset>,<shape>  for quantized types
where <type> is one of "int8q", "int16q", "int32q.
But I don`t know what this be used for.

Reply:
Per your suggestion I also added the quantized types with the format I mentioned:
-model-input=name,type,scale,offset,shape (type = int8q, int16q, int32q).
I also modified the Caffe2 unit test to use int8 quantized type.